Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,195 @@
+            const float center_screenposition_y = 0.5f*viewportSize.m_height;
+
+            // Draw title
+            const char* centerViewportDebugTextTitle = "Multiplayer Editor";
+            const float textHeight = m_fontDrawInterface->GetTextSize(m_drawParams, centerViewportDebugTextTitle).GetY();
+            const float screenposition_title_y = center_screenposition_y-textHeight*0.5f;
+            m_drawParams.m_position = AZ::Vector3(center_screenposition_x, screenposition_title_y, 1.0f);
+            m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Center;
+            m_drawParams.m_color = AZ::Colors::Yellow;
+            m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, centerViewportDebugTextTitle);
+            
+            // Draw center debug text under the title
+            // Calculate line spacing based on the font's actual line height
+            m_drawParams.m_color = AZ::Colors::White;
+            m_drawParams.m_position.SetY(m_drawParams.m_position.GetY() + textHeight + m_lineSpacing);
+            m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, m_centerViewportDebugText.c_str());
+        }
+
+        // Build the connection status string (just show client connected or disconnected status for now)
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();        
+        if (agentType == MultiplayerAgentType::Client)
+        {        
+            // Display the connection status in the bottom-right viewport
+            if (AzNetworking::INetworkInterface* networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName)))
+            {
+                AzNetworking::IConnectionSet& connectionSet = networkInterface->GetConnectionSet();
+                if (connectionSet.GetConnectionCount() > 0)
+                {
+                    connectionSet.VisitConnections([this](AzNetworking::IConnection& connection){ this->DrawConnectionStatus(connection.GetConnectionState()); });
+                }
+                else
+                {
+                    // If we're a client yet are lacking a connection then we've been unintentionally disconnected
+                    // Display a disconnect message in the viewport
+                    DrawConnectionStatus(AzNetworking::ConnectionState::Disconnected);
+                }
+            }
+        }
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::DisplayCenterViewportMessage(const char* text)
+    {
+        if (strlen(text) == 0)
+        {
+            StopCenterViewportDebugMessaging();
+            return;
+        }
+        
+        m_centerViewportDebugText = text;
+    }
+    
+    void MultiplayerConnectionViewportMessageSystemComponent::StopCenterViewportDebugMessaging()
+    {
+        m_centerViewportDebugText = "";
```

Comment:
One of these days I am going to be too lazy to see what we do with this operation? Is it the same as calling .clear() or does it create temporary objects?

Reply:
Definitely not the same as clear, ends up calling assign. Switched to use clear :P